static/script/devices/device.js

```diff
@@ -635,14 +635,58 @@ define(
              */
             loadAuthenticatedURL: function(/*url, opts*/) {
             },
+
+            /**
+             * Returns a new XMLHttpRequest. Overridden in unit tests.
+             * @returns {XMLHttpRequest} a new instance
+             * @private
+             */
+            _newXMLHttpRequest: function() {
```

Comment:
This function only seems to exist to aid unit testing - would prefer it if this wasn't necessary.

Using http://jasmine.github.io/2.0/ajax.html to mock out XMLHttpRequest might make this function unnecessary.